### PR TITLE
Improves UX which replaces ID with applications name.

### LINKED ISSUE #242: VC activity command UX - Add application name
The vc activity command text is lacking the name of the actual activity started. This is kind of confusing for others since they do not know what they will join in when clicking.

The command should mention the name of the activity and not just an id.

akvKddi

### PR DESCRIPTION
**Pull request by Tais993**
Created constants for each applications name.
Created a map for name to id convertion.
Now displays the applications name instead of ID when the user interacts with it.

Closes #242

### CODE REVIEW

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 1]
```diff
@@ -43,23 +44,50 @@
     private static final String MAX_USES_OPTION = "max-uses";
     private static final String MAX_AGE_OPTION = "max-age";
 
+
+    private static final String YOUTUBE_TOGETHER_NAME = " YouTube Together";
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:48]: A space too much at the beginning of the text

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 2]
```diff
@@ -147,17 +175,24 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
             return;
         }
 
-        OptionMapping usedOption = (applicationOption != null) ? applicationOption : idOption;
 
-        handleSubcommand(event, voiceChannel, usedOption, maxUses, maxAge);
+        String ID;
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:179]: naming convention, `id` or `applicationId`

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 3]
```diff
@@ -147,17 +175,24 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
...
+        if (applicationOption != null) {
+            ID = VC_APPLICATION_TO_ID.get(applicationOption.getAsString());
+        } else {
+            ID = idOption.getAsString();
+        }
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:184]: NIT: use ternary

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java [hunk 4]
```diff
@@ -147,17 +175,24 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
...
+
+
+        handleSubcommand(event, voiceChannel, ID, maxUses, maxAge);
     }
 
     private static void handleSubcommand(@NotNull SlashCommandEvent event,
-            @NotNull VoiceChannel voiceChannel, @NotNull OptionMapping option,
+            @NotNull VoiceChannel voiceChannel, @NotNull String id,
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:191]: NIT: `applicationId` instead of `id`

**Review by Zabuzard — CHANGES_REQUESTED** [review]

**Comment by Zabuzard** [comment]
The linked issue #242 also proposes to adjust the text message a bit to me more targeted towards the readers than the author. Right now, you only added the name of the application, it seems.

Do you also want to adjust the message a bit?

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java (hunk 2 above)
Comment (Tais993) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:179]: Probably left caps on without knowing lmao

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java (hunk 3 above)
Comment (Tais993) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java:184]: Disagree, find readability horrible due to the length of the if statement.

**Review by Zabuzard — CHANGES_REQUESTED** [review]
The linked issue #242 also proposes to adjust the text message a bit to me more targeted towards the readers than the author. Right now, you only added the name of the application, it seems.

Do you also want to adjust the message a bit?

**Review by Zabuzard — APPROVED** [review]

**Comment by Zabuzard** [comment]
Moved the other proposal into #247 , so that this is actually just #242 and not blocked anymore.

**Review by marko-radosavljevic — APPROVED** [review]

**Pull request merged**